lib/plugins/aws/invoke-local/runtime-wrappers/invoke.rb

```diff
@@ -1,44 +1,32 @@
 require 'json'
+require 'securerandom'
 
 class FakeLambdaContext
-  attr_reader :function_name, :function_version
+  attr_reader :function_name, :function_version, :aws_request_id, :log_stream_name, :memory_limit_in_mb,
+    :invoked_function_arn, :log_group_name, :deadline_ms
 
-  def initialize(name: 'Fake', version: 'LATEST', timeout: 6, **options)
-    @function_name = name
+  def initialize(function_name: 'Fake', version: 'LATEST', timeout: 6, **options)
+    @function_name = function_name
     @function_version = version
-    @created_time = Time.now()
+    @memory_limit_in_mb = 1024
     @timeout = timeout
-    options.each {|k,v|
-      send(k, v)
-    }
-  end
-
-  def get_remaining_time_in_millis
-    [@timeout*1000 - ((Time.now() - @created_time)*1000).round, 0].max
-  end
 
-  def invoked_function_arn
-    "arn:aws:lambda:serverless:#{function_name}"
-  end
+    # Allow overriding defaults
+    options.each do |k,v|
+      instance_variable_set("@#{k}", v)
+    end
 
-  def memory_limit_in_mb
-    return '1024'
-  end
+    @aws_request_id = SecureRandom.uuid
```

Comment:
why exactly would we want random ids for local invocation here?

Reply:
Because they're random in production. They're also random in the Node.js local invocation🙂. 

https://github.com/serverless/serverless/blob/5013af01476fb3d13f31775ff4171bc98a35d39d/lib/plugins/aws/invoke-local/index.js#L874

In my lambda, I attach the request ID as context to my error tracking/instrumentation service. Sometimes I enable it on local, and seeing errors with the same request ID leads to some confusing situations.

Reply:
Thanks for the clarification on this one 👍